### ISSUE TITLE
Support for nested structured fields

### ISSUE THREAD

**Issue by kubaplas** [w-vision/DataDefinitions#249]
It would be really nice if we could map nested structures like localized fields in object bricks.
That would require refactoring \Wvision\Bundle\DataDefinitionsBundle\Service\FieldSelection::getClassDefinition and appropriate setters, for example by using recursion.

| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes

**Comment by kubaplas** [comment]
@dpfaffenbauer I started working on this. The idea is as follows (for imports):
- getClassDefinition will call itself recursively on all localized fields, object bricks, blocks, fieldcolections and classification store attributes - already tested it, seems to be working
- "path" to the field will be saved as an additional field in ToColumn clas and configuration (similar to the way objectbricks currently work)
- current indentifier will stay the same, to keep backward compatibility with existing configurations
- instead of setters for fieldcollections, localized fields and objectbricks, data definition will by itself use path mentioned before to get into the field
- setters will be used only for final attributes (many-to-many relation etc).

What do you think?

**Comment by dpfaffenbauer** [comment]
I don't like the idea of the path, it doesn't feel solid enough. Since a LocalizedField in a brick should be setter wise already be accessable. "Sort of" its a combination of brick and localizedfield. What is missing is a UI to configure them, that is basically it. isn't it?

**Comment by kubaplas** [comment]
Not really, the problem arises, when we have something that requires a setter inside objectbrick or localized field (like link, objects with metadata).
Another setter could solve this one problem, but not in general. You can have localized fields in object bricks, fieldcollections and blocks (those already are 3 dedicated setters). In those localized fields you can have different fields which require different setters (like already present RelationSetter).
If we free up the setter by letting import dig into field by itself, we get more flexibility - we could use setters for replacing or appending many-to-many relations, galleries etc.

**Comment by dpfaffenbauer** [comment]
lets free up the setter by introducing another configuration similar to setter and not call it setter. So, every type gets it's specific setting like:

classificationStore: the field, group and keyId
brick: the field and class
brick localized: field, class, locale
localized field: locale

wdyt?

Since that will be a BC Break, we have to solve that within 4.0
